# Homework 4A


## Please add your team members' names here.

Team members' names

* Student Name: Cheve Garcia
* Student UT EID: gcg837

* Student Name: Eduardo Borges Issa
* Student UT EID: eb34276

## Proportion Test and Correlation


# Patient Dataset
The data in this assignment consists of body temperature measurements and heart rate measurements for 65 men and 65 women. Use this data to address the following questions.


In [1]:
# Standard Headers
# You are welcome to add additional headers here if you wish
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Enable inline mode for matplotlib so that Jupyter displays graphs
%matplotlib inline

In [2]:
df = pd.read_csv("heart-rate.csv")
df

,Temperature,Sex,HeartRate
0,96.3,M,70
1,96.7,M,71
2,96.9,M,74
3,97.0,M,80
4,97.1,M,73
...,...,...,...
125,99.3,F,68
126,99.4,F,77
127,99.9,F,79
128,100.0,F,78


## Task - 1
We are interested in whether the proportion of men and women with body temperatures greater than or equal to 98.6 degrees Fahrenheit are equal. Therefore, we need to dichotomize the body temperature variable. Create a new variable, called “temp_level” in which temp_level = 1 if body temperature >= 98.6 and temp_level=0 if body temperature < 98.6. 

**(2 points)**

In [3]:
# Add your Code Here 

df["temp_level"] = (df["Temperature"] >= 98.6).astype(int)
df

,Temperature,Sex,HeartRate,temp_level
0,96.3,M,70,0
1,96.7,M,71,0
2,96.9,M,74,0
3,97.0,M,80,0
4,97.1,M,73,0
...,...,...,...,...
125,99.3,F,68,1
126,99.4,F,77,1
127,99.9,F,79,1
128,100.0,F,78,1


## Task - 2
Summarize the data relating to body temperature level by sex.     **(4 points)**

In [4]:
# Add your Code Here 
df["temp_level"] = (df["Temperature"] >= 98.6).astype(int)
#counts & percent
counts = pd.crosstab(df["Sex"], df["temp_level"])
print(counts)
pcts = pd.crosstab(df["Sex"], df["temp_level"], normalize="index").round(3)
print(pcts)
#mean
summary = (df.groupby(["Sex", "temp_level"]).agg(n=("Temperature", "size"), mean_temp=("Temperature", "mean"), mean_hr=("HeartRate", "mean")))
print(summary)

temp_level   0   1
Sex               
F           30  35
M           51  14
temp_level      0      1
Sex                     
F           0.462  0.538
M           0.785  0.215
                 n  mean_temp    mean_hr
Sex temp_level                          
F   0           30  97.976667  73.566667
    1           35  98.965714  75.514286
M   0           51  97.729412  71.882353
    1           14  98.935714  76.642857


## Task - 3
Formally test (at the $\alpha =.05$ level) whether the proportion of people with higher body temperatures (greater than or equal to 98.6) is the same across men and women, based on this effect measure.  
Do females have higher body temperatures than males? Write your final Conclusion for the test.  **(6 points)**



In [5]:
from statsmodels.stats.proportion import proportions_ztest

#H0: mu0 = mu1 H1: mu0 != mu1
z2, p2 = proportions_ztest(count, nobs, alternative="two-sided")

#H0: mu0=mu1  H1: mu0 > mu1
z1, p1 = proportions_ztest(count, nobs, alternative="larger")

print(f"Two-sided: z = {z2:.3f}, p = {p2:.5f}")
print(f"One-sided (F > M): z = {z1:.3f}, p = {p1:.5f}")

ModuleNotFoundError: No module named 'statsmodels'

In both cases, we can reject the null hypothesis. In the two sided test, there is enough evidence to suggest that the proportion of people with a body temperature >= 98.6 differs between men and women. In the one-sided  test, there is enough evidence to suggest that females are more likely than men to have a body temperature > 98.6.

## Task - 4
What is the correlation coeffient between Heart Rate and Tempreture? Is there any correlation for different gender?  **(4 points)**

In [16]:
# Add your Code Here 
overall_corr = df['HeartRate'].corr(df['Temperature']) 
print(f'Overall Correlation between Heart Rate and Temperature: {overall_corr:.4f}')

Overall Correlation between Heart Rate and Temperature: 0.2537


In [17]:
# Add your Code Here 
male_corr = df[df['Sex'] == 'M']['HeartRate'].corr( df[df['Sex'] == 'M']['Temperature'] ) 
female_corr = df[df['Sex'] == 'F']['HeartRate'].corr( df[df['Sex'] == 'F']['Temperature'] ) 
print(f'Male Correlation: {male_corr:.4f}') 
print(f'Female Correlation: {female_corr:.4f}')

Male Correlation: 0.1963
Female Correlation: 0.2545


## Task - 5  Correlation Test $\rho$-Test 
Do a Hypothesis test to determine whether there  is a statistically significant correlation between Heart Rate and Temperature. Provide your conclusion and your p-value for the test.  **(4 points)**

In [23]:
# Add your Code Here 
from scipy import stats

# alpha = 0.05

# H0 rho = 0 (there is no linear association)
# H1 rho != 0 (there is a linear association)

r_stat, p_val = stats.pearsonr(df['Temperature'], df['HeartRate']) 
n = len(df) 
df_deg = n - 2 
alpha = 0.05

t_stat = r_stat * np.sqrt(df_deg / (1 - r_stat**2)) 
t_crit = stats.t.ppf(1 - alpha / 2, df_deg)

print(f'Pearson r: {r_stat:.4f}')
print(f't-statistic: {t_stat:.4f} (Critical t: ±{t_crit:.4f})') 
print(f'p-value: {p_val:.5f}')

Pearson r: 0.2537
t-statistic: 2.9668 (Critical t: ±1.9787)
p-value: 0.00359


At the alpha = 0.05 significance level, we reject H0 (t = 2.967, p = 0.0036$). There is statistically significant evidence of a positive linear correlation between Body Temperature and Heart Rate.